auth-tokens/src/main/java/com/palantir/tokens/auth/JsonWebToken.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tokens.auth;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.guava.GuavaModule;
+import com.google.common.io.BaseEncoding;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import org.immutables.value.Value;
+
+@Value.Immutable
+@Value.Style(visibility = Value.Style.ImplementationVisibility.PACKAGE, jdkOnly = true)
+public abstract class JsonWebToken {
+
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .registerModule(new GuavaModule());
+
+    /**
+     * Returns an unverified user id.
+     */
+    @Value.Parameter
+    public abstract String getUnsafeUserId();
```

Comment:
Ok, pushed a refactor.

We also need to add a servlet filter that pushes this into the slf4j MDC (http://logback.qos.ch/manual/mdc.html) -- so another option is to move this to that project, but I think it might be useful to key on `sid` or `jti` for cache keys.